src/flatbuffer_types.rs

```diff
@@ -174,8 +174,17 @@ impl FlatBufferManifestCodec {
             checkpoint_id: Self::decode_uuid(parent.checkpoint()),
         });
 
+        let external_dbs = manifest.external_dbs().map(|external_dbs| {
+            external_dbs.iter().map(|db| ExternalDbs {
+                path: db.path().to_string(),
+                checkpoint_id: Self::decode_uuid(db.checkpoint()),
+                sst_ids: db.sst_ids().iter().map(|id| Compacted(id.ulid())).collect(),
+            })
+        });
+
         Manifest {
             parent,
+            external_dbs: external_dbs.map(|iter| iter.collect()).unwrap_or_default(),
```

Comment:
IIUC, `external_dbs` is an iter of iters. Any reason not to `.collect()` on line 182 and 184? Seems like it might make this line a bit cleaner.

Reply:
I think the reason was solely because it broke type inference, but I'll double check tomorrow morning 🤔